lib/stream-chat/errors.rb

```diff
@@ -1,25 +1,41 @@
+# typed: strict
 # frozen_string_literal: true
 
-# lib/errors.rb
-
 module StreamChat
   class StreamAPIException < StandardError
+    extend T::Sig
+    T::Configuration.default_checked_level = :never
+    # For now we disable runtime type checks.
+    # We will enable it with a major bump in the future,
+    # but for now, let's just run a static type check.
+
+    sig { returns(Integer) }
     attr_reader :error_code
+
+    sig { returns(String) }
     attr_reader :error_message
 
+    sig { returns(T::Boolean) }
+    attr_reader :json_response
```

Comment:
Should we keep `def json_response?` if we have `attr_reader`?

Reply:
✅